hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_derivative.py

```diff
@@ -629,30 +718,64 @@ async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
-                    order_id_set.remove(result)
-                    successful_cancellations.append(CancellationResult(result, True))
-        except asyncio.TimeoutError:
-            self.logger().warning("Cancellation of all active orders for Bybit Perpetual connector"
-                                  " stopped after max wait time")
-
-        failed_cancellations = [CancellationResult(oid, False) for oid in order_id_set]
+            async with timeout(timeout_seconds):
+                cancellation_results = await safe_gather(*tasks, return_exceptions=True)
+                for cancel_result, order in zip(cancellation_results, incomplete_orders):
+                    if cancel_result and cancel_result == order.client_order_id:
+                        successful_cancellations.append(CancellationResult(order.client_order_id, True))
+                    else:
+                        failed_cancellations.append(CancellationResult(order.client_order_id, False))
+        except Exception:
+            self.logger().network(
+                "Unexpected error cancelling orders.",
+                exc_info=True,
+                app_warning_msg="Failed to cancel order with ByBit Perpetual. Check API key and network connection."
+            )
         return successful_cancellations + failed_cancellations
 
+    async def _cancel_all_account_orders(self, trading_pair: str):
+        """
+        Async cancel all account's orders, not just orders tracked by the ClientOrderTracker
+        :param trading_pair: The market (e.g. BTC-CAD) of the order.
+        """
+        try:
+            body_params = {
+                "symbol": await self._trading_pair_symbol(trading_pair),
+            }
+            response = await self._api_request(
+                method="POST",
+                endpoint=CONSTANTS.CANCEL_ALL_ACTIVE_ORDERS_PATH_URL,
+                trading_pair=trading_pair,
+                body=body_params,
+                is_auth_required=True,
+            )
+
+            response_code = response["ret_code"]
+
+            if response_code != CONSTANTS.RET_CODE_OK:
+                for order_id in list(self._client_order_tracker.active_orders.keys()):
+                    self.stop_tracking_order(order_id)
+            else:
+                raise IOError(f"Bybit Perpetual encountered a problem cancelling all account's orders "
+                              f"for {trading_pair}"
+                              f" ({response['ret_code']} - {response['ret_msg']})")
+        except Exception as e:
+            self.logger().error("Could not cancel all account orders.")
+            raise e
+
+    def cancel_all_account_orders(self, trading_pair: str):
+        """
+        Cancel all account's orders, not just orders tracked by the ClientOrderTracker
+        :param trading_pair: The market (e.g. BTC-CAD) of the order.
+        """
+        safe_ensure_future(self._cancel_all_account_orders(trading_pair))
+
```

Comment:
Ok, in that case, if changing the position mode fails due to orders being present, the strategies such as Perpetual MM would have to be blocked since they won't be allowed to cancel them by themselves.